pkg/protocol/engine/clock/blocktime/relativetime.go

```diff
@@ -81,6 +81,17 @@ func (c *RelativeTime) Advance(newTime time.Time) (updated bool) {
 	return true
 }
 
+// Reset resets the time value to the given time (resetting monotonicity of the relative time).
+func (c *RelativeTime) Reset(newTime time.Time) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
```

Comment:
Why do you lock here and not in other components' Resets?

Reply:
The "RelativeTime" is not a "component" but a self contained data structure that currently lacks the ability to just "Set" a specific time. So the "Reset" method might be used outside the scope of a reset call for the engine components simply to reset the time to a specific value that is lower than the current time, but if you think its unnecessary we can also remove it.

Reply:
I still find this discrepancy a bit odd. The Relativetime is still a subcomponent of the Clock, which is an engine-specific component. It is kinda odd that no component holds a lock while resetting but this does, either all have or none, no?

Reply:
I will add mutexes everwhere its cleaner anyway